tokio-util/src/io/poll_read_buf.rs

```diff
@@ -58,7 +58,25 @@ where
 
     let n = {
         let mut buf = ReadBuf::uninit(buf.bytes_mut());
+        let before = buf.filled().as_ptr();
```

Comment:
Could there be trouble with empty slices here?

Reply:
Hm. Depending on the implementation of `ReadBuf` this could panic if we decide to e.g. return a static empty slice for `filled` in case it's empty. At least that would be an internal decision until ReadBuf is in std. To be on the safe side it wouldn't hurt if the `ReadBuf` API guaranteed that `filled` always returns a subslice of the passed in buffer, even when empty.

I can't think of a way for it to cause a soundness issue at least.